src/main/java/net/minestom/server/adventure/ComponentHolder.java

```diff
@@ -40,4 +41,16 @@ default void visitComponents(@NotNull Consumer<Component> visitor) {
             visitor.accept(component);
         }
     }
+
+    default boolean hasTranslatableComponents() {
```

Comment:
No new API. You should move it to `PacketUtils`

You should also be aware that this method may be incredibly hot. So worth optimizing (as I see multiple expensive streams)